packages/router/router.ts

```diff
@@ -1000,15 +1000,10 @@ export function createRouter(init: RouterInit): Router {
     }
 
     if (isRedirectResult(result)) {
-      let redirectNavigation: NavigationStates["Loading"] = {
-        state: "loading",
-        location: createLocation(state.location, result.location),
-        ...submission,
-      };
```

Comment:
All redirect navigation creations are handled inside `startRedirectNavigation` now, as they're only needed for `GET` redirects